### DISCUSSION TITLE
Github API limit exceeded

### DISCUSSION THREAD

**Discussion by saviomartin** [EddieHubCommunity/support#923] [Q&A] [answered]
Hello there 👋,
I was working on Github API for my side project. I used the graphql API playground for getting the data. But, now I am seeing that API rate limit exceeded for `157.44.177.10`. How could I increase this. I have already given the token in my header.
```
headers: {
      "Content-Type": "application/json",
      Authentication: "token xxxx",
}
```
I tried to find the remaining requests for me out of 60. I get this
```
x-ratelimit-limit: 0
x-ratelimit-remaining: 0
x-ratelimit-reset: 1613380487
x-ratelimit-used: 0
content-length: 279
```
I waited for the time to get it reset. That is
```
1613380487 = 14:44:47
```
I just want to use the Github graphql API. Pls, help. Any help will be greatly appreciated. Thanks in advance 😊

**Comment by eddiejaoude** [comment, marked as answer]
The rate limit according to their docs is `5000` for authenticated users. Did you use all of this allowance already? Maybe look at optimising your application.

When it resets what number do you get?

```
"graphql": {
      "limit": 5000,
      "remaining": 4993,
      "reset": 1372700389
    },
```

https://docs.github.com/en/rest/reference/rate-limit#understanding-your-rate-limit-status

**Comment by saviomartin** [comment]
Hello @eddiejaoude ,
Thanks for your attention. I tried that `https://api.github.com/rate_limit`. This is what, I get.
```
{
  "resources": {
    "core": {
      "limit": 60,
      "remaining": 59,
      "reset": 1613392339,
      "used": 1
    },
    "graphql": {
      "limit": 0,
      "remaining": 0,
      "reset": 1613392396,
      "used": 0
    },
    "integration_manifest": {
      "limit": 5000,
      "remaining": 5000,
      "reset": 1613392624,
      "used": 0
    },
    "search": {
      "limit": 10,
      "remaining": 10,
      "reset": 1613389084,
      "used": 0
    }
  },
  "rate": {
    "limit": 60,
    "remaining": 59,
    "reset": 1613392339,
    "used": 1
  }
}
```

I had been in this problem for last 2 days. I waited to get the graphql reset but it is not resetting. What could I do next?

**Reply by eddiejaoude** [reply]
it looks like you are not authenticated, otherwise the `core` limit would be `5000` not `60`, you will see this in the docs link I shared

```
"core": {
      "limit": 5000,
      "remaining": 4999,
      "reset": 1372700873
    },
```

**Reply by saviomartin** [reply]
Ok, I got it. Maybe I would not be authorized. How could I authorize it then?? I just passed my token in the headers.
```
headers: {
      "Content-Type": "application/json",
      Authentication: "token xxxx",
}
```
Anything more should I do??

**Reply by Vyvy-vi** [reply]
maybe have a look at:
https://docs.github.com/en/rest/overview/resources-in-the-rest-api#authentication
https://docs.github.com/en/rest/overview/other-authentication-methods

**Reply by saviomartin** [reply]
Thanks now it is working, thanks for your attention and support @eddiejaoude @Vyvy-vi

**Reply by eddiejaoude** [reply]
Great 👍
